app/lib/vaccination_notification_criteria.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class VaccinationNotificationCriteria
+  def initialize(vaccination_record:)
+    @vaccination_record = vaccination_record
+  end
+
+  def call
+    patient
+      .consents
+      .where(programme:)
+      .not_invalidated
+      .not_withdrawn
+      .all?(&:notify_parents_on_vaccination)
```

Comment:
I think for consent from a parent this value will be `nil`.

Reply:
Just to add to this, I think the logic you had before was correct:

```rb
.all? { it.notify_parents_on_vaccination || it.notify_parents_on_vaccination.nil? }
```

Although I think it might read better like:

```rb
.none? { it.notify_parents_on_vaccination == false }
```